source/features/collapse-markdown-sections.ts

```diff
@@ -0,0 +1,68 @@
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+const allHeadingsSelector = 'h1, h2, h3, h4, h5, h6';
+
+function isHeading(element: HTMLElement): boolean {
+	return element.matches(allHeadingsSelector);
+}
+
+function isSameLevelHeading(referenceHeading: HTMLElement, compareHeading: HTMLElement): boolean {
+	return referenceHeading.tagName === compareHeading.tagName;
+}
+
+function isHigherLevelHeading(referenceHeading: HTMLElement, compareHeading: HTMLElement): boolean {
+	return referenceHeading.tagName.localeCompare(compareHeading.tagName) > 0;
+}
+
+function toggleSection(event: delegate.Event<MouseEvent, HTMLElement>): void {
+	let sectionHeading = (event.target as HTMLElement).closest<HTMLElement>(allHeadingsSelector)!;
+	const sectionState = sectionHeading.classList.contains('rgh-markdown-section-collapsed');
+	if (event.altKey) {
+		// Set `sectionHeading` as the first same-level header of the larger section
+		for (let element = sectionHeading; element; element = element.previousElementSibling as HTMLElement) {
+			if (!isHeading(element)) {
+				continue;
+			}
+
+			if (isHigherLevelHeading(sectionHeading, element)) {
+				break;
+			}
```

Comment:
`isSameLevelHeading` already has exact comparison, so I think you can drop this

Reply:
But really, "alt-clicking" a heading is equivalent to just clicking its parent, so perhaps it shouldn't be a thing.

I'd rather add a "View outline" button next to the "Edit Readme" so that it toggles the whole tree. However I'm not sure of where to add this button on `isSingleFile`

Reply:
> However I'm not sure of where to add this button on isSingleFile

How about next to the "Edit" button too?
![image](https://user-images.githubusercontent.com/46634000/110244710-4d930200-7f60-11eb-872f-271a980210bb.png)

Reply:
> clicking on a top-level heading (`h1`)

Some documents have multiple h1s where this wouldn't work as well but:

- eh
- you could map `click on h1` to `toggle all` specifically, rather than following the regular logic

Reply:
> you could map click on h1 to toggle all specifically, rather than following the regular logic

:+1: